iroh/src/magicsock/relay_actor.rs

```diff
@@ -98,7 +127,33 @@ impl ActiveRelayActor {
         }
     }
 
+    fn create_relay_client(
+        url: RelayUrl,
+        opts: RelayConnectionOptions,
+    ) -> (relay::client::Client, relay::client::ClientReceiver) {
+        let RelayConnectionOptions {
+            secret_key,
+            dns_resolver,
+            proxy_url,
+            prefer_ipv6,
+            #[cfg(any(test, feature = "test-utils"))]
+            insecure_skip_cert_verify,
+        } = opts;
+        let mut builder =
+            relay::client::ClientBuilder::new(url).address_family_selector(move || {
+                let prefer_ipv6 = prefer_ipv6.clone();
+                Box::pin(async move { prefer_ipv6.load(Ordering::Relaxed) })
```

Comment:
do we still need the `address_family_selector` to accept a future?

Reply:
Not sure, I was not really aiming to touch that bit of the code.  I hope when I change `relay::client::Client` to not be an actor that this would disappear, but that's for a later PR.

Reply:
I just checked, this is the only usage, I'll do a PR to change it

Reply:
https://github.com/n0-computer/iroh/pull/3044